File: BE/FleetManagementLibrary/FleetManagementLibrary/Models/Entities/RouteHistory.cs
namespace FleetManagementLibrary.Models.Entities
{
    public class RouteHistory
    {
        public long RouteHistoryID { get; set; }
        public long VehicleID { get; set; }
        public int VehicleDirection { get; set; }
        public char Status { get; set; }
        public string VehicleSpeed { get; set; }
        public long RecordTime { get; set; }
        public string Address { get; set; }
        public float Latitude { get; set; }
        public float Longitude { get; set; }
    }
}

File: FE/src/app/features/route-history/route-history.ts
import { Component } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { RouteHistoryService } from '../../services/route-history.service';

@Component({
  selector: 'route-history',
  templateUrl: './route-history.html',
})
export class RouteHistoryComponent {
  constructor(
    private route: ActivatedRoute,
    public routeHistoryService: RouteHistoryService
  ) {}
  VehicleId = Number(this.route.snapshot.paramMap.get('id'));
  fields: string[] = [
    'VehicleDirection',
    'Status',
    'VehicleSpeed',
    'Address',
    'Latitude',
    'Longitude',
    'RecordTime',
  ];
}
